code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/message/views/Message.kt

```diff
@@ -62,13 +64,38 @@ internal fun MessageScreen(
         onBackPressed()
     }
 
-    Message(
-        isVisible = presentationStateManager.visibilityState,
-        inAppMessageSettings = inAppMessageSettings,
-        gestureTracker = gestureTracker,
-        onCreated = { onCreated(it) },
-        onDisposed = { onDisposed() }
-    )
+    if (inAppMessageSettings.shouldTakeOverUi) {
+        /**
+         * Dialog is used to take over the UI when the InAppMessage is set to take over the UI.
+         * This is necessary to ensure that the InAppMessage is displayed on top of the UI.
+         * Which will ensure that ScreenReader can read the content of the InAppMessage only and not the underlying UI.
+         */
+        Dialog(
+            properties = DialogProperties(
+                usePlatformDefaultWidth = false,
+                dismissOnBackPress = true,
+                dismissOnClickOutside = false
+            ),
+            onDismissRequest = {
```

Comment:
I vaguely remember `Dialog` applying a translucent grey background automatically. How is this being avoided? Does this change honor the  case where the in-app message does not provide a background color (like a centered modal with transparent background) ?

Reply:
Removed the default dim of Dialog.

Reply:
Thanks for taking care of this!